spec/storage/fog_spec.rb

```diff
@@ -36,5 +36,16 @@
         expect(subject.filename).to eq('foo.txt')
       end
     end
+
+    context "when url contains multi-byte characters ('日本語') in query string" do
+      before do
+        # Ruby 2.3.0 is returning a string with ASCII-8BIT encoding.
```

Comment:
Thank you for your reply.

Ruby 2.2.4 is returning UTF-8 encoding, but Ruby 2.3.0 is returning ASCII-8BIT.

This `force_encoding('ASCII-8BIT')` means a multi-byte characters that encoded in ASCII-8BIT was the problem in `URI.decode`. This test code to reproduced it.

But use of `CGI.unescape` will be no longer this problem.

Would you like to delete this test code If you confusing. How you do you feel about it?